modules/importer/src/main/scala/Importer.scala

```diff
@@ -83,10 +84,26 @@ object Importer extends SafeApp {
           Nil
         )
 
+        import edu.gemini.spModel.dataset.{Dataset => OldDataset}
+        import edu.gemini.spModel.obslog.ObsLog
+
         val configs = ss.flatMap(unsafeFromConfig)
+        val obsLog  = Option(ObsLog.getIfExists(o)).fold(Map.empty[Int, OldDataset]) {
+                        _.getAllDatasetRecords.asScala.map(_.exec.dataset).map(d => d.getIndex -> d).toMap
+                      }
 
         ObservationDao.insert(newObs) *>
-        configs.zipWithIndex.traverse { case (c, n) => StepDao.insert(newObs.id, Location(n*100), c) }.void
+        configs.zipWithIndex.traverseU { case (c, n) =>
+          for {
+            id <- StepDao.insert(newObs.id, Location(n * 100), c)
+            _  <- obsLog.get(n).fold(Free.point[ConnectionOp, Unit](())) { oldDataset =>
```

Comment:
I assume there's a better/shorter way than `Free.point[ConnectionOp, Unit](())` but couldn't find it.